tests/linear_model/test_glm.py

```diff
@@ -203,3 +218,115 @@ def test_logistic_predict_proba_shape():
+
+
+@pytest.mark.parametrize("solver", ["newton", "gradient_descent"])
+@pytest.mark.parametrize("fit_intercept", [True, False])
+@pytest.mark.parametrize(
+    "reg_type, skl_reg_type, skl_params, data_generator",
+    [
+        (LinearRegression, slm.LinearRegression, {}, "medium_size_regression"),
+        (
+            LogisticRegression,
+            slm.LogisticRegression,
+            {"penalty": "none"},
+            "single_chunk_classification",
+        ),
+        (
+            PoissonRegression,
+            slm.PoissonRegressor,
+            {"alpha": 0},
+            "medium_size_counts",
+        ),
+    ],
+)
+def test_model_against_sklearn(
+    reg_type, skl_reg_type, skl_params, data_generator, fit_intercept, solver, request
+):
+    """
+    Tests accuracy of model predictions
+    """
+    X, y = request.getfixturevalue(data_generator)
+    dask_ml_model = reg_type(
+        fit_intercept=fit_intercept,
+        solver=solver,
+        penalty="l2",
+        C=1e8,
+    )
+    skl_model = skl_reg_type(fit_intercept=fit_intercept, **skl_params)
+    # skl_model has to be fit with numpy data
+    dask_ml_model.fit(X, y)
+    skl_model.fit(X.compute(), y.compute())
+    np.testing.assert_allclose(
+        dask_ml_model.predict(X), skl_model.predict(X), rtol=1e-3, atol=1e-3
+    )
+    # est, truth = dask_ml_model.predict(X), skl_model.predict(X)
+    # rel_error =
+
+
+@pytest.mark.parametrize(
+    "reg_type, skl_reg_type, skl_params, dataset_function",
+    [
+        (LinearRegression, slm.LinearRegression, {}, make_regression),
+        (
+            LogisticRegression,
+            slm.LogisticRegression,
+            {"penalty": "none"},
+            make_classification,
+        ),
+    ],
+)
+def test_model_coef_against_sklearn(
+    reg_type, skl_reg_type, skl_params, dataset_function
+):
+    """
+    Tests accuracy of coefficient calculation.
+
+    sk-learn least squares method does not tie out to iterative solvers with
+    e.g. Rdi
+    """
+    X, y = dataset_function(
+        n_samples=100, n_features=20, n_informative=10, chunks=100, random_state=0
+    )
+    X_np, y_np = X.compute(), y.compute()
+
+    skl_model = skl_reg_type(fit_intercept=True, **skl_params)
+    skl_model.fit(X_np, y_np)
+
+    dask_ml_model = reg_type(fit_intercept=True, solver="newton", penalty="l2", C=1e8)
+    dask_ml_model.fit(X, y)
+
+    np.testing.assert_allclose(
+        skl_model.coef_.flatten(), dask_ml_model.coef_, rtol=1e-3, atol=1e-8
+    )
+
+
+def test_poisson_regressor_against_sklearn(single_chunk_count_classification):
+    X, y = single_chunk_count_classification
+    skl_model = slm.PoissonRegressor(alpha=0, fit_intercept=True)
+    skl_model.fit(X, y)
+    dask_ml_model = PoissonRegression(
+        fit_intercept=True, solver="newton", penalty="l2", C=1e8
+    )
+    dask_ml_model.fit(X, y)
+    np.testing.assert_allclose(
+        skl_model.coef_.flatten(), dask_ml_model.coef_, rtol=1e-3, atol=1e-3
```

Comment:
When I look at the tests, I can't really tell if the coefficients are "close enough" with the specification of `atol` and `rtol` (e.g., what if all the coefficients are less than `atol`?).

It'd also be great if you could add some relative error condition in your other tests (it's also the test used in many optimization textbooks). This code would convince me that the coefficients are "close enough":

``` python
import numpy.linalg as LA
est = np.hstack((dask_ml_model.coef_, dask_model.intercept_))
truth = np.hstack((skl_model.coef_, slm.intercept_))

rel_error = LA.norm(est - truth) / LA.norm(truth)
assert rel_error < 1e-3
```

Reply:
That makes sense to me. I have converted all of the tests to use the relative error, as you describe above. 

There is one remaining issue that seems to be related to the gradient descent results that affects both the `PoissonRegressor` results and the `LogisticRegression` results.

First, When running the test with a relative error tolerance of `1e-3`, it typically does not pass for the `PoissonRegressor` with gradient descent (Newton's method and LBFGS achieve the same results). I have spent a while debugging the data creation functions, but have not been able to reliably get the test to pass with the `1e-3` relative error tolerance. I suspect that this is due to having a large number of features, as increasing `N / p` from `5` to `10` does decrease the error, and potentially due to a non-convex loss function for this data leading to difficulty for the gradient descent algorithm. `scikit-learn` defaults to LBFGS for `sklearn.linear_model.PoissonRegressor` with no alternative option. For the `LogisticRegression`, the test of the relative error of the coefficients using `1e-3` is similarly too large to pass (as the test is written, it is approx. `0.14`) when using gradient descent. It converges to the `scikit-learn` results for both Newton's method and LBFGS.

Maybe the best strategy is to remove gradient descent from the tests and supply some sort of convergence warning when using vanilla GD?

I appreciate all of the time that you have put into looking over these results!